cyipopt/scipy_interface.py

```diff
@@ -555,20 +550,50 @@ def minimize_ipopt(fun,
                           njev=problem.njev,
                           nit=problem.nit)
 
+
 def _minimize_ipopt_iv(fun, x0, args, kwargs, method, jac, hess, hessp,
                        bounds, constraints, tol, callback, options):
     # basic input validation for minimize_ipopt that is not included in
     # IpoptProblemWrapper
-
-    x0 = np.asarray(x0)[()]
+    x0 = np.atleast_1d(x0)
     if not np.issubdtype(x0.dtype, np.number):
         raise ValueError('`x0` must be a numeric array.')
 
     if method is not None:  # this will be updated when gh-200 is merged
         raise NotImplementedError('`method` is not yet supported.`')
 
-    # TODO: add input validation for `bounds` when adding
-    #  support for instances of new-style constraints (e.g. `Bounds`)
+    # Handle bounds that are either sequences (of sequences) or instances of
+    # `optimize.Bounds`.
+    if bounds is None:
+        bounds = [-np.inf, np.inf]
+
+    if isinstance(bounds, optimize.Bounds):
+        lb, ub = bounds.lb, bounds.ub
+    else:
+        bounds = np.atleast_2d(bounds)
+        if bounds.shape[1] != 2:
+            raise ValueError("`bounds` must specify both lower and upper "
+                             "limits for each decision variable.")
+        lb, ub = bounds.T
+
+    try:
+        lb, ub, x0 = np.broadcast_arrays(lb, ub, x0)
```

Comment:
Somewhat related to the comment above: here, I'm making the function even more flexible about input shape. Previously, the function would raise an error if the user passes only a single lower and upper bound but there are many decision variables; this broadcasts the bounds to apply to all the decision variables.

`scipy.optimize.minimize` is more strict than this, but it doesn't do a great job of validating the input, so you get some imperfect error messages. It's also sort of a mess in this department, so I'm not sure if it's been thought through very carefully. 

LMK if you'd prefer to stay strict and require the number of bounds to equal the number of decision variables rather than following broadcasting rules.

Reply:
As long as it works and is tested you can make it flexible.